numba/tests/test_stencils.py

```diff
@@ -365,12 +407,1272 @@ def test_impl(n):
+
+    # Should this work at all? does it make sense?
+    # pyStencil raises as there's no rel indices
+    def test_basic25(self):
+        """1d idx on 2d arr"""
+        a = np.arange(12).reshape(3, 4)
+
+        def kernel(a):
+            return 1.
+        self.check(kernel, a, expected_exception=[ValueError])
+
+    def test_basic26(self):
+        """3d arr"""
+        a = np.arange(64).reshape(4, 8, 2)
+
+        def kernel(a):
+            return a[0, 0, 0] - a[0, 1, 0] + 1.
+        self.check(kernel, a)
+
+    def test_basic27(self):
+        """4d arr"""
+        a = np.arange(128).reshape(4, 8, 2, 2)
+
+        def kernel(a):
+            return a[0, 0, 0, 0] - a[0, 1, 0, -1] + 1.
+        self.check(kernel, a)
+
+    def test_basic28(self):
+        """type widen """
+        a = np.arange(12).reshape(3, 4).astype(np.float32)
+
+        def kernel(a):
+            return a[0, 0] + np.float64(10.)
+        self.check(kernel, a)
+
+    def test_basic29(self):
+        """const index from func """
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, int(np.cos(0))]
+        self.check(kernel, a, expected_exception=[ValueError, LoweringError])
+
+    def test_basic30(self):
+        """signed zeros"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            return a[-0, -0]
+        self.check(kernel, a)
+
+    # this passes parfors, fails in pyStencil
+    def test_basic31(self):
+        """does a const propagate?"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            t = 1
+            return a[t, 0]
+        self.check(kernel, a)
+        
+    # this fails all, need to decide whether to entirely support const prop + fold
+    # or not.
+    def test_basic31b(self):
+        """does a const propagate?"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            s = 1
+            t = 1 - s
+            return a[t, 0]
+        self.check(kernel, a)
+
+    def test_basic32(self):
+        """typed int index"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            return a[np.int8(1), 0]
+        self.check(kernel, a, expected_exception=[ValueError, LoweringError])
+
+    def test_basic33(self):
+        """add 0d array"""
+        a = np.arange(12.).reshape(3, 4)
+
+        def kernel(a):
+            return a[0, 0] + np.array(1)
+        self.check(kernel, a)
+
+    def test_basic34(self):
+        """More complex rel index with dependency on addition rel index"""
+        def kernel(a):
+            g = 4. + a[0, 1]
+            return g + (a[0, 1] + a[1, 0] + a[0, -1] + np.sin(a[-2, 0]))
+        a = np.arange(144).reshape(12, 12)
+        self.check(kernel, a)
+
+    # parfors ought to raise but does not, is this likely down to the allocation of the
+    # return array under parfors being a different type?
+    def test_basic35(self):
```

Comment:
Think this should be fixed so behaviour is consistent?

Reply:
Fixed.